compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -1605,69 +1606,110 @@ LogicalResult WinogradOutputTransformOp::getResultTilePosition(
 //===----------------------------------------------------------------------===//
 
 SmallVector<Range> AttentionOp::getIterationDomain(OpBuilder &builder) {
-  int64_t iterationDomainRank = getIterationDomainRank();
-  SmallVector<Range> loopBounds(iterationDomainRank);
+  AttentionOpDetail opInfo(*this);
+  ArrayRef<AffineMap> indexingMaps = opInfo.getIndexingMaps();
+
+  SmallVector<Range> loopBounds(opInfo.getDomainRank());
   Location loc = getLoc();
   Value zero = builder.create<arith::ConstantIndexOp>(loc, 0);
   Value one = builder.create<arith::ConstantIndexOp>(loc, 1);
-  Value source = getQuery();
-  for (auto dim : llvm::seq<int64_t>(0, iterationDomainRank)) {
+
+  for (auto dim : llvm::seq<int64_t>(0, opInfo.getDomainRank())) {
     loopBounds[dim].offset = zero;
-    loopBounds[dim].size = getDimValue(builder, loc, source, dim);
     loopBounds[dim].stride = one;
   }
+
+  SmallVector<bool> dimsFound(opInfo.getDomainRank(), false);
+  auto fillSizes = [&](Value val, AffineMap indexingMap) {
+    for (auto [idx, dimExpr] : llvm::enumerate(indexingMap.getResults())) {
+      assert(isa<AffineDimExpr>(dimExpr));
+      AffineDimExpr dim = cast<AffineDimExpr>(dimExpr);
+      int64_t pos = dim.getPosition();
+      if (dimsFound[pos]) {
+        continue;
+      }
+      dimsFound[pos] = true;
+      loopBounds[pos].size = getDimValue(builder, loc, val, idx);
+    }
+  };
+
+  // Sizes can be found from Q, K, V alone.
+  fillSizes(getQuery(), indexingMaps[0]);
+  fillSizes(getKey(), indexingMaps[1]);
+  fillSizes(getValue(), indexingMaps[2]);
```

Comment:
Instead of inferring them from indexingMap (and do we really need indexing maps?), can we make them fixed? Different from linalg op, LinalgExt ops have more strict definition. I'm worried that we are making logics hard to follow while it could just be simple.

Reply:
Eventually, we want to have indexing_maps on linalg_ext op so we can do fusions with that op. A simple example is transpose fused with attention op.